docker-compose-separate-containers.yml

```diff
@@ -1,23 +1,25 @@
-version: '2'
+version: '3'
 services:
   nginx:
     image: nginx
     container_name: nginx
     ports:
       - "80:80"
     volumes:
-      - /etc/nginx/conf.d
+      - nginx_conf:/etc/nginx/conf.d
 
   dockergen:
     image: jwilder/docker-gen
     command: -notify-sighup nginx -watch /etc/docker-gen/templates/nginx.tmpl /etc/nginx/conf.d/default.conf
-    volumes_from:
-      - nginx
     volumes:
       - /var/run/docker.sock:/tmp/docker.sock:ro
       - ./nginx.tmpl:/etc/docker-gen/templates/nginx.tmpl
+      - nginx_conf:/etc/nginx/conf.d
```

Comment:
1. Build an `nginx` image with your own configuration files in `/etc/nginx/conf.d`.
2. Delete the containers and the volume.
3. Start the `dockergen` service.
4. Start the `nginx` service.

You'll see that the `nginx-conf` volume was initialized to empty from the `dockergen` image. Then it'll only have one `default.conf` after generation from template. Your own configuration files would be lost.